### DISCUSSION TITLE
couple of questions

### DISCUSSION THREAD

**Discussion by mehrdad2000** [tilln/jmeter-iso8583#102] [General]
Hi
I've had couple of questions:

1-Need to continuously listen to specific port to response message, as background process.

here is my groovy response script, when I run it, will wait for 60 second then stopped:

responder.groovy 
message.setResponseMTI()
message.set(39, '00')
source.send(message)

2-I've enabled debug mode and able to see unpacked message that received but can't see full response that generate by groovy script and send back.

here is the log:
`2024-08-12 15:15:11,184 DEBUG n.c.b.j.i.Q2: (channel/192.168.1.1:36296) [send] Out: 0110 192292 00091119
`

Any idea?
Thanks

**Comment by tilln** [comment]
> when I run it, will wait for 60 second then stop

If you are running in Server Mode there may be no incoming connection on that port.
That's the most likely cause for the 60 second wait.
See [`jmeter.iso8583.incomingConnectionTimeout`](https://github.com/tilln/jmeter-iso8583?tab=readme-ov-file#jmeter-properties)

Check the logs for 60 lines like the following
`INFO n.c.b.j.i.ISO8583Config: Waiting 60 seconds for incoming client connection`

> can't see full response that generate by groovy script

I suggest you `dump` the message as part of your groovy script, either to `System.out` or to the log like so:

```groovy
def baos = new java.io.ByteArrayOutputStream()
def ps = new java.io.PrintStream(baos)
message.dump(ps, '')
log.info(baos.toString())
```

**Reply by mehrdad2000** [reply]
@tilln yes i’m in server mode, is there any way to keep it up without timeout? Available unlimitedly?

**Reply by tilln** [reply]
> @tilln yes i’m in server mode, is there any way to keep it up without timeout? Available unlimitedly?

Well, I don't see how that would make sense. If there is no incoming connection ever, why would you run in server mode in the first place?
